### PR TITLE
*: move kv to server/core.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: server/cache_test.go [hunk 1]
```diff
@@ -568,3 +568,31 @@ func newMockIDAllocator() *mockIDAllocator {
 func (alloc *mockIDAllocator) Alloc() (uint64, error) {
 	return atomic.AddUint64(&alloc.base, 1), nil
 }
+
```
Comment (disksing) [inline server/cache_test.go]: following 2 funcs are moved from deleted `kv.go`.

File: server/config.go [hunk 2]
```diff
@@ -477,8 +478,29 @@ func (o *scheduleOption) RemoveSchedulerCfg(name string) error {
...
+func (o *scheduleOption) persist(kv *core.KV) error {
+	cfg := &Config{
+		Schedule:    *o.load(),
+		Replication: *o.rep.load(),
+	}
+	err := kv.SaveConfig(cfg)
+	return errors.Trace(err)
+}
+
+func (o *scheduleOption) reload(kv *core.KV) error {
```
Comment (disksing) [inline server/config.go]: `reload` is previously a method of server. see https://github.com/pingcap/pd/pull/781/files#diff-2c671b5ee7007449d6115af14481769eL370

File: server/core/kv.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+// You may obtain a copy of the License at
+//
+//	   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
```
Comment (disksing) [inline server/core/kv.go]: Move from server/kv.go, with some methods exported.

File: server/etcd_kv.go [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+// You may obtain a copy of the License at
+//
+//	   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
```
Comment (disksing) [inline server/etcd_kv.go]: Move from server/kv.go.

**Comment by siddontang** [comment]
Please fix the conflict. @disksing

**Pull request merged**